src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -309,14 +304,33 @@ public Task CopyToAsync(Stream stream, TransportContext context)
                 throw new ArgumentNullException("stream");
             }
 
-            TaskCompletionSource<object> tcs = new TaskCompletionSource<object>();
+            TaskCompletionSource<object> tcs = new TaskCompletionSource<object>(this);
             try
             {
                 Task task = null;
                 if (IsBuffered)
                 {
-                    byte[] data = this.GetDataBuffer(_bufferedContent);
-                    task = stream.WriteAsync(data, 0, (int)_bufferedContent.Length);
+                    _bufferedContent.Position = 0;
```

Comment:
Why use CopyToAsync and continuations at all?  Why can't this just use TryGetBuffer? e.g.
```C#
ArraySegment<byte> data;
if (!_bufferedContent.TryGetBuffer(out data))
    data = _bufferedContent.ToArray();
task = stream.WriteAsync(data.Array, data.Offset, data.Count);
```
Do we expect _bufferedContent to typically be read-only such that TryGetBuffer would frequently fail, or something like that?